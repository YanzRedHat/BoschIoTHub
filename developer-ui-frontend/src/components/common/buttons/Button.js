/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import styled from "styled-components";

// This is the base styled component for a Button which has to be extended
// by the implementation
const Button = styled.button`
  cursor: pointer;
  font-size: 1.6rem;
  border-radius: 0.2rem;
  padding: 0.5rem 2rem;
  svg {
    display: inline-block;
    width: auto;
  }
  &:disabled {
    cursor: not-allowed;
    color: ${props => props.theme.disabledFontColor};
  }
`;

export default Button;
