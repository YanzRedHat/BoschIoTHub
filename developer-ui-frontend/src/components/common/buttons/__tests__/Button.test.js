/*
 * Copyright 2018 Bosch Software Innovations GmbH ("Bosch SI"). All rights reserved.
 */
import React from "react";
import { shallow } from "enzyme";
import Button from "components/common/buttons/Button.js";

describe("Button", () => {
  it("should be defined", () => {
    expect(Button).toBeDefined();
  });
  it("should render correctly", () => {
    const tree = shallow(<Button name="button test" />);
    expect(tree).toMatchSnapshot();
  });
});
