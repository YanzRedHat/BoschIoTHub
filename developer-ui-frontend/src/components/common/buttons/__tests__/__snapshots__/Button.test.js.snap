// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<button
  className="sc-bdVaJa sklfR"
  name="button test"
/>
`;
